File: 01 Two Sums/twoSums_bruteForce.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    
    //Declare and initialise a dynamic array to store returned results of size two

    //Dereference the returnSize pointer and initialise its value to 2
    *returnSize = 2;

    // Declaration of a dynamic array
    int* result = malloc(*returnSize * sizeof(int));

    // Check if malloc has failed. NULL if malloc failed.
    if (result == NULL)
    {
        return 1;
    }

    // Iterate over the array and compare addition of two values to target. Since loop starts at zero, i and j will loop until it's numsSize-1 so i < numsSize or i <= numsSize - 1
    for (int i = 0; i < numsSize ; i++)
    {
        // make sure that the jth index is not compared to ith index
        for (int j = i + 1 ; j < numsSize ; j++)
        {
            // if the ith index and ith index value adds up to target, store value in result array.
            if (nums[i] + nums[j] == target)
            {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }
    // After looping through the array, and there is no two sums that adds up to target, *returnSize = 0
    *returnSize = 0;
    free(result);
    return 0;
}

File: 01 Two Sums/twoSums_bruteForce.py
# Version 1 - Created list as a separate entity
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # Create a list to store indices of 2 number that adds up to target
        result = []

        # A way to obtain indices from nums such as the 2 nums adds up to target

        # the i index starts at 0
        for i in range(len(nums)):
            # the j index starts at j+1 using the pyton range(start, stop) type
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    # Append indices to list to list
                    result.append(i)
                    result.append(j)
                    return result
                
        return result